src/mainwindow.cpp

```diff
@@ -488,20 +495,11 @@ void MainWindow::on_actionBeta_log_directory_triggered()
 
 void MainWindow::on_actionChoose_directory_triggered()
 {
-    // Use file path of the current tab as default directory.
-    // It works with both file and directory paths.
-    QString defaultDir;
-    TreeModel* model = GetCurrentTreeModel();
-    if (model)
-    {
-        LogTab* currentTab = m_logTabs[model];
-        defaultDir = currentTab->GetTabPath();
-    }
-
-    QString directoryPath = QFileDialog::getExistingDirectory(this, "Select directory to monitor", defaultDir);
+    QString directoryPath = QFileDialog::getExistingDirectory(this, "Select directory to monitor", m_lastOpenFolder);
```

Comment:
does this even if you haven't opened a folder ever before (brand new user scenario). What directory gets opened in the file dialog?